backend/routes.py

```
@@ -558,11 +860,125 @@ def generate_headings(data: dict):
+            # Handle different storage types securely
+            if temp_storage_type == "memory" and content_hash and job_id:
+                # Try to retrieve from memory storage
+                memory_storage = get_memory_storage()
+                file_bytes = memory_storage.retrieve(f"{job_id}_{content_hash}")
+                if file_bytes:
+                    print(
+                        f"Retrieved file content from secure memory storage (hash: {content_hash[:8]}...)"
+                    )
+                else:
+                    print(
+                        f"Warning: File content not found in memory storage, may have been cleaned up"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)